src/webgl/context-state.js

```diff
@@ -511,50 +91,67 @@ function getState(gl, {copyState = false} = {}) {
 
 // GETTERS AND SETTERS
 
-/**
- * Sets value with key to context.
- * Value may be "normalized" (in case a short form is supported). In that case
- * the normalized value is retured.
- *
- * @param {WebGLRenderingContext} gl - context
- * @param {String} key - parameter name
- * @param {*} value - parameter value
- * @return {*} - "normalized" parameter value after assignment
- */
+export function getParameterDefaultValue(gl, key) {
+  // Get the parameter definition for the key
+  const parameterDefinition = GL_PARAMETERS[key];
+  assert(parameterDefinition, key);
+
+  // Return the default value
+  return parameterDefinition.value;
+}
+
+// Get the parameter value(s) from the context
+// Might return an array of multiple values.
+// The return value will be acceptable input to setParameter
 export function getParameter(gl, key) {
-  const parameterDefinition = GL_STATE[key];
-  if (!parameterDefinition) {
-    throw new Error(`Unknown GL state parameter ${key}`);
-  }
-  // Get the parameter value(s) from the context
+  // Get the parameter definition for the key
```

Comment:
Yes caching is only partly implemented. It would require some additional work and test cases to make it into luma v4.

Hoping to collaborate with @1chandu on this.